eth/db/chain.py

```diff
@@ -314,27 +216,27 @@ def add_transaction(self,
 
     def get_block_transactions(
             self,
-            header: BlockHeader,
-            transaction_class: Type['BaseTransaction']) -> Iterable['BaseTransaction']:
+            header: BlockHeaderAPI,
+            transaction_class: Type[SignedTransactionAPI]) -> Tuple[SignedTransactionAPI, ...]:
         """
         Returns an iterable of transactions for the block speficied by the
         given block header.
         """
         return self._get_block_transactions(header.transaction_root, transaction_class)
 
-    def get_block_transaction_hashes(self, block_header: BlockHeader) -> Iterable[Hash32]:
```

Comment:
And I found a few places like this where our return types were allowing for things like returning a generator when that is typically a foot-gun.